### PR TITLE
feat(api,ui): add workflow usage and fix some bugs

### PR DESCRIPTION
**Pull request by bnjjj**
Signed-off-by: Benjamin Coenen <benjamin.coenen@corp.ovh.com>

### CODE REVIEW

File: ui/src/assets/i18n/fr.json [hunk 1]
```diff
@@ -892,6 +892,7 @@
   "worker_model_shell_docker": "Commande shell",
   "worker_model_post_cmd": "Commande après exécution du worker",
   "worker_model_shell_cmd": "Commande principale du shell",
+  "worker_model_no_usage": "Ce worker modèle n'est pas utilisé explicitement dans un pipeline",
```
Comment (fsamin) [inline ui/src/assets/i18n/fr.json]: modèle de worker

File: engine/api/pipeline/pipeline.go [hunk 2]
```diff
@@ -94,6 +94,28 @@ func LoadPipelineByID(ctx context.Context, db gorp.SqlExecutor, pipelineID int64
 	return &p, nil
 }
 
+// LoadByWokerModelName loads pipelines from database for a given worker model name
+func LoadByWokerModelName(db gorp.SqlExecutor, workerModelName string) ([]sdk.Pipeline, error) {
```
Comment (richardlt) [inline engine/api/pipeline/pipeline.go]:
```suggestion
func LoadByWorkerModelName(db gorp.SqlExecutor, workerModelName string) ([]sdk.Pipeline, error) {
```

File: engine/api/pipeline/pipeline.go [hunk 3]
```diff
@@ -94,6 +94,28 @@ func LoadPipelineByID(ctx context.Context, db gorp.SqlExecutor, pipelineID int64
 	return &p, nil
 }
 
+// LoadByWokerModelName loads pipelines from database for a given worker model name
```
Comment (richardlt) [inline engine/api/pipeline/pipeline.go]:
```suggestion
// LoadByWorkerModelName loads pipelines from database for a given worker model name
```

File: engine/api/pipeline/pipeline.go [hunk 4]
```diff
@@ -94,6 +94,28 @@ func LoadPipelineByID(ctx context.Context, db gorp.SqlExecutor, pipelineID int64
...
+		FROM action_requirement
+			JOIN pipeline_action ON action_requirement.action_id = pipeline_action.action_id
+			JOIN pipeline_stage ON pipeline_action.pipeline_stage_id = pipeline_stage.id
+			JOIN pipeline ON pipeline.id = pipeline_stage.pipeline_id
+			JOIN project ON project.id = pipeline.project_id
+		WHERE action_requirement.type = 'model' AND action_requirement.value = $1`
+
+	if _, err := db.Select(&pips, query, workerModelName); err != nil {
+		if err == sql.ErrNoRows {
+			return pips, nil
```
Comment (richardlt) [inline engine/api/pipeline/pipeline.go]:
```suggestion
			return nil, nil
```

**Pull request merged**